r-package/grf/R/forest_summary.R

```diff
@@ -115,69 +122,93 @@ test_calibration <- function(forest) {
 #' forest <- causal_forest(X, Y, W)
 #' best_linear_projection(forest, X[,1:2])
 #' }
-#' 
+#'
 #' @return An estimate of the best linear projection, along with coefficient standard errors.
 #'
 #' @importFrom stats lm
-#' 
+#'
 #' @export
-best_linear_projection <- function(forest, A = NULL, subset = NULL) {
-  
-  cluster.se <- length(forest$clusters) > 0
-  
+best_linear_projection <- function(forest,
+                                   A = NULL,
+                                   subset = NULL,
+                                   num.trees.for.variance = 500,
+                                   seed = runif(1, 0, .Machine$integer.max)) {
```

Comment:
For the user who wants more control, let's do the following. Instead of letting them give a `seed`, let them provide a full vector of `debiasing.weights`. Notice that the variance forest is only used for computing `debiasing.weights`, so if these weights are provided (or computed externally), we don't need to run a variance forest in the first place (and so seed is no longer needed).

[Note: For consistency, `debiasing.weights` should be a vector of length `n`; then we subset it like everything else.]